Include step by step guide (GSI-476) (#31)
ghga-de/ghga-datasteward-kit: README.md

Validation of the metadata can be achieved using the
[GHGA Metadata Validator](https://github.com/ghga-de/ghga-validator/).

Once your spreadsheet or JSON file have passed validation, you may send the metadata to
the Central Data Steward.

#### B. File Encryption and Upload:

This is achieved using the data steward kit, using the following steps:

1. **Generate credentials**: The kit interacts with services at GHGA central. To
   authenticate yourself against these services you need to create a set of credentials
   using the `ghga-datasteward-kit generate-credentials` command. Please see
   [this section](#generate-catalog-accessions) for further details.

2. **Encrypt and Upload**: File encryption and upload to the S3-compatible object
   storage is done in one go. This is achieved using either the
   `ghga-datasteward-kit files upload` for uploading a single file or the
   `ghga-datasteward-kit files batch-upload` for uploading multiple files at once.
   Please see [this section](#files-batch-upload) for further details. This will output
   one summary JSON per uploaded file. The encryption secret is automatically

>    transferred to GHGA central.

Author: This documentation is mostly targeted at Data Stewards not at DevOps people. The DevOps people will provide the configuration details to the Data Stewards, this is mentioned in the config docs.

Once the upload of all files of a submission has completed, please notify the GHGA
Central Data Steward and provide the summary JSONs obtained in step 2.

### Central Data Steward

> The responsible for ingesting the metadata and the upload summary files into the running
> system. This is performed with the following steps:

Reviewer: ```suggestion
The person responsible for ingesting the metadata and the upload summary files into the running
system. This is performed using the the following steps:
```

Author: Ah this should rather be 
```suggestion
The Central Data Steward is responsible for ingesting the metadata and the upload summary files into the running
system. This is performed with the following steps:
```

1. **Generate credentials**: As for the local data stewards, central data stewards need
   to have credentials for authentication with GHGA Central Services. To create these
   credentials the `ghga-datasteward-kit generate-credentials` command is used.
   Please see [this section](#generate-catalog-accessions) for further details.
2. **Transpile Metadata Spreadsheet**: If the Local Data Steward provided a metadata
   spreadsheet, this spreadsheet has to be first transpiled to the JSON format using the
   `ghga-datasteward-kit metadata transpile` command. Please see
   [this section](#metadata) for further details.
3. **Include in Submission Registry**: The submission JSON is included in a
   submission registry on the local file system using the
   `ghga-datasteward-kit metadata submit` command. Please see
   [this section](#metadata) for further details.
4. **Produce Metadata Artifacts**: A transformation workflow is run on all submissions
   in the submission registry to produce multiple query-specific metadata artifacts
   using the `ghga-datasteward-kit metadata transform` command. Please see
   [this section](#metadata) for further details.

> 5. **Publish Metadata**: To publish all metadata artifacts of all submission to the

Reviewer: ```suggestion
5. **Publish Metadata**: To publish all metadata artifacts of all submissions to the
```

Attention: Keep this output file in a safe, private location.
If this file is lost, the uploaded file content becomes inaccessible.

### files ingest-upload-metadata

> *To be performed by Central Data Stewards.*

Reviewer: ```suggestion
*To be performed by Central Data Stewards only.*
```

Upload all file summary JSONs (produced using the
[files (batch-)upload](#files-batch-upload) command) from the given directory to the
running system and make the corresponding files available for download.

This command requires a configuration file as described [here](./ingest_config.md).

### metadata

*To be performed by Central Data Stewards only.*

The metadata label groups metadata related commands.

Some of them require a configuration file as described [here](./metadata_config.md).

### load

*To be performed by Central Data Stewards only.*

The load command makes files and metadata available to user in the running system.

It needs a configuration parameters as described [here](./load_config.md).

### generate-catalog-accessions

A command to generate a token/hash pair for interacting with GHGA Central services.

The generated token file should not be moved to a different system and never be shared
with another user.
The token hash (**not the token*) must be shared with the GHGA Central Operation
Team. This process has to be done only once per data steward and system (if a data
steward is working with multiple compute environments, one set of credentials per

> environment should be created).

Reviewer: This seems to be in the wrong subsection (generate-catalog accessions)

Author: You are write, I misread. Will fix

Author: I competely removed the generate-catalog-accessions from the readme because it does not add to the help that is provided when running --help. Moreover, this is a legacy endpoint that should not be that prominent (You only need it when you already know of it).
